build.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cugraph build script
+
+# This script is used to build the component(s) in this repo from
+# source, and can be called with various options to customize the
+# build as needed (see the help output for details)
+
+# Abort script on first error
+set -e
+
+NUMARGS=$#
+ARGS=$*
+
+# NOTE: ensure all dir changes are relative to the location of this
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean libcugraph cugraph -v -g -n -h --help"
+HELP="$0 [<target> ...] [<flag> ...]
+ where <target> is:
+   clean      - remove all existing build artifacts and configuration (start over)
+   libcugraph - build the cugraph C++ code
+   cugraph    - build the cugraph Python package
+ and <flag> is:
+   -v         - verbose build mode
+   -g         - build for debug
+   -n         - no install step
+   -h         - print this text
+
+ default action (no args) is to build and install 'libcugraph' then 'cugraph' targets
+"
+LIBCUGRAPH_BUILD_DIR=${REPODIR}/cpp/build
+CUGRAPH_BUILD_DIR=${REPODIR}/python/build
+BUILD_DIRS="${LIBCUGRAPH_BUILD_DIR} ${CUGRAPH_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+
+# Set defaults for vars that may not have been defined externally
+#  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
+#         CONDA_PREFIX, but there is no fallback from there!
+INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}
+PARALLEL_LEVEL=${PARALLEL_LEVEL:=""}
```

Comment:
Should this default to the number of CPU core of the machine? cugraph build takes a while otherwise.

Reply:
This is given by `nproc` or `cat /proc/cpuinfo | grep processor | wc -l`

Reply:
The default results in a `make -j`, which builds in parallel with "no limit", and was the default for the other repos. I can change it to use `nproc` if you prefer though.